src/xcode/ENA/ENA/Source/Scenes/ContactDiary/AddAndEditEntry/DiaryAddAndEditEntryViewModel.swift

```diff
@@ -5,16 +5,121 @@
 import Foundation
 import UIKit
 
-struct DiaryAddAndEditEntryViewModel {
+class DiaryAddAndEditEntryViewModel {
+
+	// MARK: - Init
+
+	init(
+		mode: Mode,
+		diaryService: DiaryService,
+		dismiss: @escaping () -> Void
```

Comment:
What is the benefit of routing the user interaction through the view model?
The closure can be just part of the ViewController.

Also from the calling point of view. Calling "viewModel.dismiss()".
What is dismissed from the viewmodel? Some data? It sounds a little bit strange.

Reply:
Reworked it